carddav/client.go

```diff
@@ -429,3 +429,52 @@ func (c *Client) PutAddressObject(path string, card vcard.Card) (*AddressObject,
 	}
 	return ao, nil
 }
+
+// SyncCollection do a sync-collection operation on resource(path), it returns a SyncResponse
+func (c *Client) SyncCollection(path string, query *SyncQuery) (*SyncResponse, error) {
+	var limit *internal.Limit
+	if query.Limit > 0 {
+		limit = &internal.Limit{NResults: uint(query.Limit)}
+	}
+
+	propReq, err := encodeAddressPropReq(&query.DataRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	ms, err := c.ic.SyncCollection(path, query.SyncToken, internal.DepthOne, limit, propReq)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &SyncResponse{SyncToken: ms.SyncToken}
+	for _, resp := range ms.Responses {
+		p := resp.Hrefs[0].Path
```

Comment:
There is `Response.Path` which takes care of this logic. `internal.IsNotFound` can be used on the returned error to figure out whether the resource has been deleted.

Reply:
the reason I chose AddressObject is that I thought it make more sense to return a similar and structured interface

Reply:
No, properties are never returned for deleted objects.